src/mflux_debugger/coverage_report.py

```
@@ -0,0 +1,1082 @@
+        # Find dead lines
+        dead_lines = [line for line in executable_lines if line not in executed_lines]
+
+        # Find branches
+        branches = self._find_branches(tree, file_path, executed_lines)
+
+        # Find unused classes and functions
+        unused_definitions = self._find_unused_definitions(tree, file_path, executed_lines)
+
+        return dead_lines, branches, unused_definitions
+
+    def _find_executable_lines(self, tree: ast.AST, source: str) -> List[int]:
```

Ruff (0.14.3)

Unused method argument: `source`

(ARG002)